AzureDataLake/DecisionServiceExtractor/HeaderOnly.cs

```diff
@@ -116,7 +116,7 @@ private static int CountArrayElements(JsonTextReader jsonReader)
                         case JsonToken.Integer:
                             numActions++;
                             break;
-                        case JsonToken.EndObject:
+                        case JsonToken.EndArray:
```

Comment:
this will work for "a" (the actions), but not always for "o": (observations as they can be complex). really one needs to count the nesting.

other than that looks good.

Reply:
`JsonToken.EndObject` was not working for "a": it was not stopping at the end of array overcounting the number of actions.
Currenlty, this code is never invoked for the "o" field, only for "a" field.
What do you have in mind for `case JsonToken.EndObject:` in the "o" case?
"o" is a list of json objects. What is the meaning of counting its integers fieds?